CONTRIBUTING.md

```diff
@@ -30,9 +38,49 @@ The following are required to build the devfile index container image containing
     - This will also validate the devfiles in this repository, making sure they conform to a minimum standard.
     - This step will also be run in the PR build for the repository.
 
-5) (Optional) Push the container image to a container registry and [deploy](https://github.com/devfile/registry-support#deploy) a custom devfile registry on to Kubernetes or OpenShift. Test odo with the devfile stack hosted in that registry.
-  
-    - For now, strictly an optional requirement, but recommended.
-    - PR tests will also verify this.
+5) Open a pull request against this repository with a brief description of the change.
+
+
+### Updating
+
+Updating an existing devfile stack is relatively straightforward:
+
+1) Find the stack under the `stacks/` folder that you wish to update.
+2) Make the necessary changes to the stack, such as: updating image tags, commands, starter projects, etc.
+3) Update the version of stack, following the [semantic versioning format](https://semver.org/).
+4) Test your changes:
+    
+    - Minimally, testing with odo (`odo create`, `odo push`, etc) is recommended, however if your Devfile is used with other tools, it's recommended to test there as well.
+5) Open a pull request against this repository with your changes.
+
+## Samples
+
+### Contributing
+
+The devfile samples used in this devfile registry are stored in the `extraDevfileEntries.yaml` file in the root of the repository. To add a devfile sample:
```

Comment:
I think the purpose of `extraDevfileEntries.yaml` is for letting user add remote stack/sample to the registry instead of only for adding samples, that is also why we have top level attribute stack/sample in the file. Given that, if we still divide the docs sections into stack and sample, I think we may need to mention the `extraDevfileEntries.yaml` in stack section to add remote stack into the registry.

Reply:
@GeekArthur We don't really support remote stacks at this moment (in terms of registry bootstrapping), so I deliberately left it out.